src/common/BlogApi.res

```diff
@@ -81,6 +81,49 @@ let getAllPosts = () => {
   })
 }
 
+let getLivePosts = () => {
+  let postsDirectory = Node.Path.join2(Node.Process.cwd(), "_blogposts")
+
+  let livePosts = mdxFiles(postsDirectory)->Js.Array2.map(path => {
+    let {GrayMatter.data: data} =
+      Node.Path.join2(postsDirectory, path)->Node.Fs.readFileSync->GrayMatter.matter
+    switch BlogFrontmatter.decode(data) {
+    | Error(msg) => Js.Exn.raiseError(msg)
```

Comment:
Do we really need to raise here? I would just convert to option and use Belt.Array.keepMap. The error can be logged though.

Reply:
I have just copy-pasted it from the existing code above. but yeah let's a look how we can improve it :)

Reply:
Hmm... Wouldn't it be better for the build to fail early than to lose a post once published due to a format mistake? I do that on all the important content sites.

Of course,  collecting multiple errors and reporting at once is much nicer. But that doesn't seem to make much sense now that all operations are implemented as blocking sync.

I can switch all tasks to be concurrent in another PR. I wonder how much that speeds up the entire build.

Reply:
Right, my frontend-focused brain has me fooled. Keep it as it is. Refactors can be done in a separate PR.